frontend/src/screens/Profile/Profile.tsx

```diff
@@ -0,0 +1,174 @@
+import { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import styles from "./Profile.module.css";
+import WorkoutTemplateCard from "../../components/WorkoutTemplateCard/WorkoutTemplateCard";
+import SessionHistory from "../../components/SessionHistory/SessionHistory";
+import followIcon from "../../assets/followIcon.svg";
+import unfollowIcon from "../../assets/unfollowIcon.svg";
+
+const Profile = () => {
+  const { userId } = useParams<{ userId: string }>();
+  const [user, setUser] = useState<User | null>(null);
+  const [isFollowing, setIsFollowing] = useState(false);
+  const [isCurrentUser, setIsCurrentUser] = useState(false);
+
+  useEffect(() => {
+    fetchUserProfile();
```

Comment:
Instead of 3 separate fetch requests that execute serially, can we send the 3 network requests concurrently?